modules/ServerAPI/src/client/Protocol.coffee

```diff
@@ -802,21 +803,33 @@ class EndpointListController extends AbstractFormController
 			@$(".bv_endpointColumnConcentrationUnits").remove()
 			@$(".bv_concentrationUnitsPickListParent").remove()
 			@$(".bv_concentrationInputParent").remove()
-
-	getExperimentSummaryTable: =>
-		#hide previously shown warnings/success text
-		@$(".bv_downloadSuccess").hide()
-		@$(".bv_downloadWarning").hide()
-
+	
+	getExperimentsForProtocol: => 
+		# Get all the experiments associated with the protocol
 		protocolCode = @model.escape('codeName')		
 		$.ajax
 			type: 'GET'
 			#there are two similar routes in ExperimentBrowserRoutes.coffee and ExperimentServiceRoutes.coffee
 			#url: "/api/experimentsForProtocol/#{protocolCode}" #ExperimentBrowserRoutes.coffee route
 			url: "/api/experiments/protocolCodename/#{protocolCode}" #ExperimentServiceRoutes.coffee route
 			success: (experiments) =>
-				@setupExperimentSummaryTable experiments
-				@$(".bv_experimentTableControllerTitle").html "Experiments using " + protocolCode + ":"
+				#save the experiments so we don't have to retrieve them again 
+				@protocolExperiments = experiments
```

Comment:
experiments here is an array of javascript objects. It would simplify a lot of your loops through the states and values later on in the code to turn these javascript objects into a Backbone collection of Experiment Backbone objects.

You can change this line to:
```
@protocolExperiments = new ExperimentList experiments
```

Then later on in the code like on line 906 you can use `@protocolExperiments.filter` with experiment methods like `dataColumnOrderState = experiment.get("lsStates").getStatesByTypeAndKind "metadata", "data column order"`

And if the experiment has that state, you can then call:
```
dataColumnOrderState.getValuesByTypeAndKind("column name")
```

Reply:
Done - thanks for the explanation! Knowing this will be really helpful for me going forward.